### PR TITLE
feat: add hyeth flashmint

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Adds flash minting for hyETH in trade flow

## TODOs
- [ ] test with real nav price once endpoint is updated

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/app/leverage/provider.tsx [hunk 1]
```diff
@@ -339,7 +339,6 @@ export function LeverageProvider(props: { children: any }) {
       if (!provider || !publicClient) return
       if (inputTokenAmount <= 0) return
       if (!indexToken) return
-      const jsonRpcProvider = publicClientToProvider(publicClient)
```
Comment (janndriessen) [inline src/app/leverage/provider.tsx:342]: only one instance in the project left after this 🙌 

@0xonramp check my proposal here https://github.com/IndexCoop/index-app/issues/1332 to get rid of it entirely in the near future.

File: src/lib/utils/api/zeroex.ts [hunk 2]
```diff
@@ -7,9 +7,13 @@ import {
...
 
-export function getConfiguredZeroExApi(swapPathOverride: string): ZeroExApi {
+export function getConfiguredZeroExSwapQuoteProvider(
+  chainId: number,
+): ZeroExSwapQuoteProvider {
+  const networkKey = getNetworkKey(chainId)
+  const swapPathOverride = `/${networkKey}/swap/v1/quote`
   const headers = getIndexApiHeaders()
-  return new ZeroExApi(
+  return new ZeroExSwapQuoteProvider(
```
Comment (janndriessen) [inline src/lib/utils/api/zeroex.ts:16]: ZeroExSwapQuoteProvider is configured just as `ZeroExApi` before.

_In the future that's where we'd plug-in a LiFiSwapQuoteProvider once it's created._

File: src/lib/hooks/use-best-quote/utils/flashmint.ts [hunk 3]
```diff
@@ -49,18 +49,15 @@ async function getEnhancedFlashMintQuote(
...
+    const swapQuoteProvider = getConfiguredZeroExSwapQuoteProvider(chainId)
     const request = {
       isMinting,
       inputToken: { ...inputToken, address: inputTokenAddress },
       outputToken: { ...outputToken, address: outputTokenAddress },
       indexTokenAmount,
       slippage,
     }
-    const quoteProvider = new FlashMintQuoteProvider(jsonRpcProvider, zeroExApi)
+    const quoteProvider = new FlashMintQuoteProvider(rpcUrl, swapQuoteProvider)
```
Comment (janndriessen) [inline src/lib/hooks/use-best-quote/utils/flashmint.ts:60]: Most changes are just satisfy the new constructor requirement for the `FlashMintQuoteProvider`

**Review by 0xonramp — COMMENTED** [review]
LGTM 👍

**Pull request merged**